### ISSUE TITLE
Installation error 

### ISSUE THREAD

**Issue by yogeshbalaji** [pytorch/pytorch#2062]
I am installing pytorch from source using python 2.7. I am getting the following error:

 File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/gen.py", line 6, in <module>
    import preprocess_declarations
  File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/preprocess_declarations.py", line 3, in <module>
    from function_wrapper import TYPE_FORMAL_GENERIC
  File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/function_wrapper.py", line 2, in <module>
    from code_template import CodeTemplate
  File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/code_template.py", line 13, in <module>
    class CodeTemplate(object):
  File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/code_template.py", line 15, in CodeTemplate
    '(^[^\n\S]*)?\$([^\d\W]\w*|\{,?[^\d\W]\w*\,?})', re.MULTILINE)
  File "/scratch0/Softwares/virtualpython/env_new/lib64/python2.7/re.py", line 190, in compile
    return _compile(pattern, flags)
  File "/scratch0/Softwares/virtualpython/env_new/lib64/python2.7/re.py", line 242, in _compile
    raise error, v # invalid expression
sre_constants.error: nothing to repeat

CMake Error at CMakeLists.txt:126 (message):
  Failed to get generated_cpp list

-- Configuring incomplete, errors occurred!

**Comment by lynic** [comment]
Might be related to recently merge pr https://github.com/pytorch/pytorch/pull/2033 I guess.

**Comment by soumith** [comment]
@yogeshbalaji what is your cmake version? `cmake --version`

**Comment by yogeshbalaji** [comment]
@soumith  cmake version 3.8.1

**Comment by killeent** [comment]
@yogeshbalaji what is your Python version?

**Comment by soumith** [comment]
maybe @yogeshbalaji has a debug build of python? Alban sent this PR to fix something that looks similar; https://github.com/pytorch/pytorch/pull/2065

**Comment by killeent** [comment]
I'm thinking it may be related to: https://stackoverflow.com/questions/3675144/regex-error-nothing-to-repeat given that the line here:

```
File "/scratch0/Softwares/pytorch_new/torch/lib/ATen/code_template.py", line 15, in CodeTemplate
'(^[^\n\S])?$([^\d\W]\w|{,?[^\d\W]\w*,?})', re.MULTILINE)
```

Has a regex string that matches this error case, which apparently was fixed in newer versions of Python.

**Comment by killeent** [comment]
Yeah, I cannot repro on both `13.6.1` and `2.7.13`

**Comment by yogeshbalaji** [comment]
@killeent The version I am using is Python 2.7.5, however I'm running it in virtualenv

**Comment by soumith** [comment]
@yogeshbalaji maybe that's the issue, prob fixed in `2.7.12` / `2.7.13`

**Comment by soumith** [comment]
it does say: `This bug seems to have been fixed between 2.7.5 and 2.7.6.` on the stackoverflow page

**Comment by yogeshbalaji** [comment]
The python version was indeed the issue. Worked when I used Python 2.7.12.
Thanks everyone for the help.
